File: php/LineUp/module/module_command.php
<?php
include(dirname(__FILE__)."/exec.php");
include(dirname(__FILE__)."/LuModule.php");
include(dirname(__FILE__)."/../Version.php");


/* 
 * Click nbfs://nbhost/SystemFileSystem/Templates/Licenses/license-default.txt to change this license
 * Click nbfs://nbhost/SystemFileSystem/Templates/Scripting/EmptyPHP.php to edit this template
 */

$module = new ModuleSystem();

function load_module($module_folder, $file) {
    global $module, $modules_version;
    foreach (json_decode(file_get_contents($file), true) as $folder) {
        if (is_dir($module_folder."/".$folder)) {
            $mod = json_decode(file_get_contents($module_folder."/".$folder."/packages.json"), true);
            $module_can = false;
            foreach ($mod["packages-id"]["interpreter-version"] as $version) {
                if (in_array($version, $modules_version)) {
                    $module_can = true;
                    break;
                }
            }
            if ($module_can) {
                $module->add_module($mod["packages-name"]);
                include($module_folder."/".$folder."/".$mod["php-files"]["file"]);
                foreach ($mod["php-files"]["class-name"] as $class_name) {
                    $module->add_class($mod["packages-name"], $class_name, new $class_name());
                }
            }
        }
    }
}

function decode_modules() {
    global $module;
    $data = array();
    foreach ($module as $module_name => $module_data) {
        foreach ($module_data as $class_name => $class_data) {
            $data[$module_name."-".$class_name] = $class_data;
        }
    } 
    return $data;
}

File: LineUp/python/LineUp/module/module_command.py
from .lumodule import ModuleSystem
import os
import runpy
import json
from .command import Exec
from ..version import modules_version


module = ModuleSystem()


def load_modules(module_folder, file):
    for folder in json.load(open(file)):
        if os.path.isdir(os.path.join(module_folder, folder)):
            mod = json.load(open(os.path.join(module_folder, folder, "packages.json")))
            module_can = False
            for i in mod["packages-id"]["interpreter-version"]:
                if i in modules_version:
                    module_can = True
                    break
            if module_can:
                module.add_modules(mod["packages-name"])
                data = runpy.run_path(os.path.join(module_folder, folder, mod["py-files"]["file"]),
                                      init_globals={"Exec": Exec, "module": module})
                for class_name in mod["py-files"]["class-name"]:
                    module.add_class(mod["packages-name"], class_name, data[class_name])


def decode_modules():
    data = {}
    for i in module.module_name.keys():
        for k, v in module.module_name[i].class_data.items():
            data[i+"-"+k] = v
    return data
